src/plugins/aws/ssm/index.ts

```diff
@@ -266,6 +268,34 @@ const spawnSsmNode = async (
     });
   });
 
+/**
+ * Spawns a child process to add a private key to the ssh-agent. The SSH agent is included in the OpenSSH suite of tools
+ * and is used to hold private keys during a session. The SSH agent typically does not persist keys across system reboots
+ * or logout/login cycles. Once you log out or restart your system, any keys added to the SSH agent during that session
+ * will need to be added again in subsequent sessions.
+ */
+const executeScpCommand = async (
+  credential: AwsCredentials,
+  command: string,
+  privateKey: string
+) => {
+  // Execute should not leave any ssh-agent processes running after it's done performing an SCP transaction.
+  const execute = `
+  eval $(ssh-agent) >/dev/null 2>&1
+  trap 'kill $SSH_AGENT_PID' EXIT
```

Comment:
I'm not sure we have to take on the job of running and killing the ssh agent. Most of the time it will already be running, and in that case the user will not expect that p0 stops it.

We could check if it's running and ask the user to run it if it's not. I tried and if I rerun `eval (ssh-agent)` when it's already running it will spawn a new process, presumably with empty ssh-key state.

Then we could add and remove by trapping `ssh-add -d ~/.ssh/sshkeynamewithout.pub`? Not sure that would work.

Reply:
> I'm not sure we have to take on the job of running and killing the ssh agent. Most of the time it will already be running, and in that case the user will not expect that p0 stops it.

I mostly did this to prevent the `ssh-agent` DOS and to prevent sharing keys with other running agents.

If the user has a bunch of other keys in the `ssh-agent` it will try to use each key to make a connection request. I found that having 5+ unrelated keys blocks me out of the host because it'll think I'm trying to break by attempting to connect many times in a split second frequently.

We won't kill any other running agents, only the ssh-agent we spawn get's killed, running `eval $(ssh-agent)` spawns a new agent and since we're running as a child process we shouldn't mess with any other process' environment  variables